sparta/specializer/funtional/batch_matmul.py

```diff
@@ -0,0 +1,189 @@
+        if forward_kernel is not None:
+            if self._biased:
+                self.forward_C = lambda A, B, bias: forward_kernel(A, B, bias)
+            else:
+                self.forward_C = lambda A, B: forward_kernel(A, B)
+            if self._mode == 'dds' and self._compressed:
+                C_converter = self._kernels['forward:C'].get_converter('C')
+                self.backward_bias = lambda grad_C: C_converter.sum(grad_C, axis=-2)
+            else:
+                self.backward_bias = lambda grad_C: grad_C.sum(-2)
+        backward_A_kernel = self._kernels['backward:A'].active_kernel()
+        if backward_A_kernel is not None:
+            if self._transpose_A:
+                self.backward_A = lambda grad_C, B: backward_A_kernel(B, grad_C)
+            else:
+                self.backward_A = lambda grad_C, B: backward_A_kernel(grad_C, B)
+        backward_B_kernel = self._kernels['backward:B'].active_kernel()
+        if backward_B_kernel is not None:
+            if self._transpose_B:
+                self.backward_B = lambda grad_C, A: backward_B_kernel(grad_C, A)
+            else:
+                self.backward_B = lambda grad_C, A: backward_B_kernel(A, grad_C)
+
+    def set_sample_inputs(
+        self,
+        sample_inputs: List[torch.Tensor],
+        sample_grads: Optional[List[torch.Tensor]] = None,
+    ):
+        A = sample_inputs[0]
+        B = sample_inputs[1]
+        if 'A' in self._masks:
+            A = A.detach() * self._masks['A']
+        elif 'B' in self._masks:
+            B = B.detach() * self._masks['B']
+        if self._biased:
+            bias = sample_inputs[2].detach()
+            self._kernels['forward:C'].set_sample_inputs([A, B, bias])
+        else:
+            self._kernels['forward:C'].set_sample_inputs([A, B])
+        if sample_grads is not None:
+            grad_C = sample_grads[0]
+            if 'C' in self._masks:
+                grad_C = grad_C.detach() * self._masks['C']
+            if self._transpose_A:
+                self._kernels['backward:A'].set_sample_inputs([B, grad_C])
+            else:
+                self._kernels['backward:A'].set_sample_inputs([grad_C, B])
+            if self._transpose_B:
+                self._kernels['backward:B'].set_sample_inputs([grad_C, A])
+            else:
+                self._kernels['backward:B'].set_sample_inputs([A, grad_C])
+
+    def get_connections(self, backward: bool = False):
+        if self._compressed and backward:
+            conditions = [{}, {}]
+            for kernel_name, tesa_shapes in self._tesa_shapes.items():
+                for k, dim in enumerate(tesa_shapes):
+                    conditions[k][kernel_name] = f'BLOCK_SIZE_{dim}_VALUE'
+            return conditions
+        else:
+            return []
+
+    def dense_forward(self, *args):
+        return self._kernels['forward:C'].dense_func(*args)
+
+
+class SparseBatchMatMulFunc(torch.autograd.Function):
```

Comment:
please double check whether this function behaves like the sparse version of [torch.bmm](https://pytorch.org/docs/stable/generated/torch.bmm.html)
if yes, please give some docstring like 
> If `input` is a (b×n×m) tensor, `mat2` is a (b×m×p) tensor, `out` will be a (b×n×p) tensor.
> out_i = input_i @ mat2_i